### ISSUE TITLE
Error building package xerces-c:x64-windows 

### ISSUE THREAD

**Issue by seanwarren** [Microsoft/vcpkg#2019]
Building on Windows 7SP1 x64 with Visual Studio 2017 15.4

```
  Package: xerces-c:x64-windows
  Vcpkg version: 0.0.91-2017-10-11-7cd465b8897bd9487db6d990e58dcac472b3d4d6
```

Fails with the following error log:

```
     1>Project "C:\vcpkg\buildtrees\xerces-c\src\xerces-c-Xerces-C_3_1_4\projects\Win32\VC14\xerces-all\xercesLib\xercesLib.vcxproj" on node 1 (Rebuild target(s)).
     1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\bin\Microsoft.Common.CurrentVersion.targets(94,27): error MSB4086: A numeric comparison was attempted on "$(TargetPlatformVersion)" that evaluates to "﻿10.0.16299.0" instead of a number, in condition "'$(UseOSWinMdReferences)' == '' and ('$(TargetPlatformWinMDLocation)' == '' and '$(TargetPlatformIdentifier)' == 'Windows' and '$(TargetPlatformVersion)' > '7.0')". [C:\vcpkg\buildtrees\xerces-c\src\xerces-c-Xerces-C_3_1_4\projects\Win32\VC14\xerces-all\xercesLib\xercesLib.vcxproj]
     1>Done Building Project "C:\vcpkg\buildtrees\xerces-c\src\xerces-c-Xerces-C_3_1_4\projects\Win32\VC14\xerces-all\xercesLib\xercesLib.vcxproj" (Rebuild target(s)) -- FAILED.
```

**Comment by seanwarren** [comment]
$(TargetPlatformVersion) appears to have a "ZERO WIDTH NO-BREAK SPACE", unicode character 0xFEFF before the number (10.0.16299.0), I don't know if that could cause any problems

**Comment by oZed** [comment]
It looks to be a problem with the getWindowsSDK.ps1, I saved the output from the script to file and saw the same 0xFEFF character at the start.  I did manage to get tbb and libusb to build by either commenting out the `/p:WindowsTargetPlatformVersion=${_csc_TARGET_PLATFORM_VERSION}` line in vcpkg_build_msbuild.cmake or replacing ${_csc_TARGET_PLATFORM_VERSION} with 10.0.16299.0

**Comment by ras0219-msft** [comment]
Should be fixed by #2034. Thanks for reporting this issue!
